### PR TITLE
Upgrade syntax with pyupgrade + code inspections + flake8

### PR DESCRIPTION
**Pull request by hugovk**
* Upgraded Python syntax with [pyupgrade](https://github.com/asottile/pyupgrade).
* Fixed some code inspections.
* Fixed some flake8 warnings.

### CODE REVIEW

**Comment by hugovk** [comment]
I wonder why the CI isn't building PRs?

**Comment by hugovk** [comment]
Passing build: https://travis-ci.org/hugovk/pyroma/builds/511687027

**Comment by regebro** [comment]
Thanks, but since classifiers.py is actually autogenerated updating it like this doesn't make sense, it's the code that autogenerates it that needs to be fixed. I'll deal with this later, as these changes are mostly cosmetic.

**Comment by hugovk** [comment]
Sure, thanks!

File: pyroma/pypidata.py [hunk 1]
```diff
@@ -32,7 +31,7 @@ def get_data(project):
         releases = [p['version'] for p in reversed(projects)]
     release = releases[0]
     # Get the metadata:
-    logging.debug("Found %s version %s" % (project, release))
+    logging.debug("Found {} version {}".format(project, release))
```
Comment (vpoulailleau) [inline pyroma/pypidata.py]: In fact, according to logging documentation, we should not do:

```python
logging.debug("Found %s version %s" % (project, release))
```
nor

```python
logging.debug("Found {} version {}".format(project, release))
```
But we should do:

```python
logging.debug("Found %s version %s", project, release)
```

The main difference here is the fact that logging is smart enough to do the formatting only when it's needed. With the two not recommended code, if the logging level is, say, info level, the debug message are not treated, but the formatting is still done.

Comment (hugovk) [reply pyroma/pypidata.py]: Interesting, please could you share a link to the logging docs?

Comment (vpoulailleau) [reply pyroma/pypidata.py]: This is not an exhaustive comment, but here are some links:

https://docs.python.org/3/library/logging.html#logrecord-objects:

> msg – The event description message, possibly a format string with placeholders for variable data.

https://docs.python.org/3/library/logging.html#logging.debug:

> The msg is the message format string, and the args are the arguments which are merged into msg using the string formatting operator.

https://www.python.org/dev/peps/pep-0282/:

> In particular, localization and formatting (which are relatively expensive) are deferred until the Handler requests them.

https://docs.python.org/3/howto/logging.html#logging-variable-data:

> To log variable data, use a format string for the event description message and append the variable data as arguments.

https://docs.python.org/3/howto/logging.html#loggers:

> The message is actually a format string, which may contain the standard string substitution syntax of %s, %d, %f, and so on. The rest of their arguments is a list of objects that correspond with the substitution fields in the message.

https://docs.python.org/3/howto/logging.html#optimization:

> Formatting of message arguments is deferred until it cannot be avoided. 

https://docs.python.org/3/howto/logging-cookbook.html#use-of-alternative-formatting-styles:

> One thing to note is that you pay no significant performance penalty with this approach: the actual formatting happens not when you make the logging call, but when (and if) the logged message is actually about to be output to a log by a handler.

**Comment by cthoyt** [comment]
The lines to fix to make the classifiers.py look nicer when it's generated are

https://github.com/regebro/pyroma/blob/f2e4ce6f7d8149ef5e273ffad91ece0837ce7df4/fetch_classifiers.py#L49-L58

**Comment by regebro** [comment]
Thanks! 

I've reformatted the code with `black` for consistency. Will check with pep8 shortly.

**Pull request closed without merging**